### DISCUSSION TITLE
Onboarding: allow me to select "system theme" instead of light or dark

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#872] [Ideas]
### What feature would you like to see?

On the "Make yourself at home" page, I have to pick light or dark mode. I have macOS set up to switch between modes at night and during the day, and I want my browser to match my system color scheme.

If this determines how the browser treats `prefers-color-scheme` media queries on websites, this is an even higher priority request. If websites will still honor my system setting then it's less urgent.

### What platform are you seeing the problem on?

macOS

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/258_

**Comment by SomeDudeBrosky** [comment]
This seems more like a bug than a feature request

**Comment by christian-bendiksen** [comment]
This is a bug on Gnome and KDE too. Changing from light to dark theme and vice versa does not automatically change the theme in Zen.

**Comment by schmidsi** [comment]
<img width="1582" alt="Screenshot 2024-12-04 at 12 25 14" src="https://github.com/user-attachments/assets/ba05053c-920a-41a2-bcd2-f7b02f482b42">

It seems like there is a dedicated theme called "System Theme" that automatically switches.

**Reply by ryanhiebert** [reply]
This was just the hint I needed, but the most obvious setting is a different one that is apparently ignored if the theme (which defaulted to "dark" for me) is set to anything but "System theme - auto" as described above. That makes this feel like a bug to me, that the main setting isn't respected in the default theme. I venture to suggest that the best default isn't dark theme, it's system, but a suboptimal default theme wouldn't be a bug.

<img width="684" alt="Screenshot 2025-01-31 at 8 36 40 AM" src="https://github.com/user-attachments/assets/ff514bea-b564-4277-88cd-23830fd79fd0" />

**Reply by ryanhiebert** [reply]
Or, at least, the way that these settings interact is not intuitive to me. I intuited the "automatic" setting to be based on system theme, but it's actually based on the theme, which is not given the same prominence as this setting is. The choice to follow the system theme seems to me like one that needs to have higher billing than a particular theme setting override.

**Reply by dceluis** [reply]
While here, are instructions on how to trigger the 'System theme' to change from light to dark documented anywhere? I tried changing the GTK theme to dark mode and that did not work. What configuration does Zen recognize to determine using the light or dark theme?

**Reply by Vann06-2005** [reply]
> <img alt="Screenshot 2024-12-04 at 12 25 14" width="1582" src="https://private-user-images.githubusercontent.com/103280/392355288-ba05053c-920a-41a2-bcd2-f7b02f482b42.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDY2NjY4NDcsIm5iZiI6MTc0NjY2NjU0NywicGF0aCI6Ii8xMDMyODAvMzkyMzU1Mjg4LWJhMDUwNTNjLTkyMGEtNDFhMi1iY2QyLWY3YjAyZjQ4MmI0Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUwOFQwMTA5MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wM2E0NjdkYjgyMzkyODlhMjhiYzYzMWEyNjA5N2M2MTdhNWZhNzE4MzEyY2RjYmFjN2M0MjA1ZjJjMzUxYmQ5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6JaBBpBG9wiU9UM3hg2_8O141Cr_76Q22t0a92qZVxs">
> 
> It seems like there is a dedicated theme called "System Theme" that automatically switches.

It work

**Comment by Dedsecproject** [comment]
I just stumbled upon this issue in macos. While I had modified the `"gfx.webrender.compositor.force-enabled"` in **about:config** page. I noticed that it worked but if I changed workspace it would break and go back to what it was before. That is until I noticed this post and went to modify the themes to enable the **_system theme - auto._** and voila, issue resolved. Not sure why themes overrides scheme preferences. that should not be the way it works. themes should only modify the way the panels of the browser looks while respecting the light mode or dark mode on webistes.

**Reply by dougdoner** [reply]
I agree, maybe the default behavior should be to have theme system theme - auto selected?

**Comment by koichiio** [comment]
I set `layout.css.prefers-color-scheme.content-override` to 3, even though GPT was suggesting that I set it from 0 to 2, and it just works. I didn't dig into how it works, and I'm not familiar with Firefox, but this is for those of you who ended up landing on this page

**Reply by thecountrox** [reply]
This is the only thing that works for me, thank you.

**Reply by scottrepreneur** [reply]
Solved it for me also
